### PR TITLE
Fix: GitHub Workflow Fails to Publish to `NPM`

### PR DESCRIPTION
**Pull request by wesdevpro**
**The Problem**: The NPM publish workflow failed to run because no `package-lock.json` was present in the root directory. This should have been easily generated by running `npm install`. However, there were some dependency issues involving `eslint` and some in house dependencies used for configuring `eslint` in the `nuxt-buefy` project. 

**The Presented Solution (in this PR)**: 
1. Bump our in house `eslint-config-buefy` to the latest version (`v0.0.3`)
2. Roll back `eslint` in the `nuxt-buefy` project to `v4.18.2` to solve the conflict to allow `npm install` to generate the `package-lock.json` file
3. Add a step to the GitHub Publish Workflow to generate the `package-lock.json` in the event it is not pushed to GitHub.

### CODE REVIEW

**Comment by wesdevpro** [comment]
@kikuomax @gmercey [Circle CI Build Error](https://app.circleci.com/pipelines/github/buefy/nuxt-buefy/113/workflows/5d27e79b-058c-4942-86b1-fb6f29f97d59/jobs/183)

Now our `Circle Ci` tests are failing when the following command is run `NODE_OPTIONS=--openssl-legacy-provider yarn test`

#### The Logs
```
yarn run v1.22.19
$ npm run lint && jest

> nuxt-buefy@0.4.26 lint
> eslint lib test

 FAIL  test/module.test.js
  ● Test suite failed to run



    Vue packages version mismatch:

    - vue@2.7.14
    - vue-server-renderer@2.7.10

    This may cause things to work incorrectly. Make sure to use the same version for both.

      at Object.<anonymous> (node_modules/vue-server-renderer/index.js:8:9)
      at Object.<anonymous> (node_modules/@nuxt/vue-renderer/dist/vue-renderer.js:19:27)
      at Object.<anonymous> (node_modules/@nuxt/server/dist/server.js:19:21)

----------|----------|----------|----------|----------|-------------------|
File      |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
----------|----------|----------|----------|----------|-------------------|
All files |  Unknown |  Unknown |  Unknown |  Unknown |                   |
----------|----------|----------|----------|----------|-------------------|
Handlebars: Access has been denied to resolve the property "statements" because it is not an "own property" of its parent.
You can add a runtime option to disable the check or this warning:
See https://handlebarsjs.com/api-reference/runtime-options.html#options-to-control-prototype-access for details
Handlebars: Access has been denied to resolve the property "branches" because it is not an "own property" of its parent.
You can add a runtime option to disable the check or this warning:
See https://handlebarsjs.com/api-reference/runtime-options.html#options-to-control-prototype-access for details
Handlebars: Access has been denied to resolve the property "functions" because it is not an "own property" of its parent.
You can add a runtime option to disable the check or this warning:
See https://handlebarsjs.com/api-reference/runtime-options.html#options-to-control-prototype-access for details
Handlebars: Access has been denied to resolve the property "lines" because it is not an "own property" of its parent.
You can add a runtime option to disable the check or this warning:
See https://handlebarsjs.com/api-reference/runtime-options.html#options-to-control-prototype-access for details
Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        0.827s
Ran all test suites.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

Exited with code exit status 1
```

### Some Questions
1. Why are we using yarn for testing/development when `Buefy/Buefy` is using `NPM`?
2. Why are the test now failing

**Comment by kikuomax** [comment]
@wesdevpro 
I have investigated the issue, the following error was solved by upgrading `nuxt`:
```
Vue packages version mismatch:

    - vue@2.7.14
    - vue-server-renderer@2.7.10
```
The latest `nuxt` compatible with Vue 2.x was `v2.17.1`. After upgrading `nuxt`, the test failed with another error:
```
 FAIL  test/module.test.js
  ● Test suite failed to run

    Cannot find module 'node:util' from 'consola.deac7d5a.cjs'

      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:221:17)
      at Object.<anonymous> (node_modules/consola/dist/shared/consola.deac7d5a.cjs:3:19)
```
This error suggested that `consola` package was using a prefixed Node package (`node:*`) that was introduced since Node v14.18 and v16, but `jest-resolve` did not understand it. So I also bumped `jest` to `v29.7.0`, which I felt too new though, and the test passed.

Bonus: we no longer need `NODE_OPTIONS=--openssl-legacy-provider` workaround.

File: .github/workflows/npm-publish.yml [hunk 1]
```diff
@@ -17,6 +17,7 @@ jobs:
           node-version: '16.x'
           registry-url: 'https://registry.npmjs.org/'
           scope: 'next-buefy'
+      - run: npm install
```
Comment (kikuomax) [inline .github/workflows/npm-publish.yml:20]: @wesdevpro I think this line is no longer necessary, because we now include `package-lock.json`.

**Comment by wesdevpro** [comment]
> ## [Codecov](https://app.codecov.io/gh/buefy/nuxt-buefy/pull/139?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=buefy) Report
> > Merging [#139](https://app.codecov.io/gh/buefy/nuxt-buefy/pull/139?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=buefy) ([f4ad6f0](https://github.com/buefy/nuxt-buefy/commit/f4ad6f0c98018ae249babdea5cba16d6e725a2bb)) into [master](https://app.codecov.io/gh/buefy/nuxt-buefy/commit/f4cc5fddf55d142bb8caa331dc45005ce6f2c13d?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=buefy) ([f4cc5fd](https://github.com/buefy/nuxt-buefy/commit/f4cc5fddf55d142bb8caa331dc45005ce6f2c13d)) will **decrease** coverage by `10.00%`.
> > The diff coverage is `n/a`.
> 
> ```diff
> @@             Coverage Diff              @@
> ##            master     #139       +/-   ##
> ============================================
> - Coverage   100.00%   90.00%   -10.00%     
> ============================================
>   Files            1        2        +1     
>   Lines            2       20       +18     
>   Branches         0        5        +5     
> ============================================
> + Hits             2       18       +16     
> - Misses           0        2        +2     
> ```
> 
> [see 1 file with indirect coverage changes](https://app.codecov.io/gh/buefy/nuxt-buefy/pull/139/indirect-changes?src=pr&el=tree-more&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=buefy)
> 
> 📣 We’re building smart automated test selection to slash your CI/CD build times. [Learn more](https://about.codecov.io/iterative-testing/?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=buefy)

@kikuomax what is this?

**Comment by kikuomax** [comment]
> > ## [Codecov](https://app.codecov.io/gh/buefy/nuxt-buefy/pull/139?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=buefy) Report
> > > Merging [#139](https://app.codecov.io/gh/buefy/nuxt-buefy/pull/139?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=buefy) ([f4ad6f0](https://github.com/buefy/nuxt-buefy/commit/f4ad6f0c98018ae249babdea5cba16d6e725a2bb)) into [master](https://app.codecov.io/gh/buefy/nuxt-buefy/commit/f4cc5fddf55d142bb8caa331dc45005ce6f2c13d?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=buefy) ([f4cc5fd](https://github.com/buefy/nuxt-buefy/commit/f4cc5fddf55d142bb8caa331dc45005ce6f2c13d)) will **decrease** coverage by `10.00%`.
> > > The diff coverage is `n/a`.
> > 
> > 
> > ```diff
> > @@             Coverage Diff              @@
> > ##            master     #139       +/-   ##
> > ============================================
> > - Coverage   100.00%   90.00%   -10.00%     
> > ============================================
> >   Files            1        2        +1     
> >   Lines            2       20       +18     
> >   Branches         0        5        +5     
> > ============================================
> > + Hits             2       18       +16     
> > - Misses           0        2        +2     
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > [see 1 file with indirect coverage changes](https://app.codecov.io/gh/buefy/nuxt-buefy/pull/139/indirect-changes?src=pr&el=tree-more&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=buefy)
> > 📣 We’re building smart automated test selection to slash your CI/CD build times. [Learn more](https://about.codecov.io/iterative-testing/?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=buefy)
> 
> @kikuomax what is this?

@wesdevpro I don't know. I followed the link "see 1 file ...", and it did not look a big deal.

**Review by kikuomax — APPROVED** [review]
@wesdevpro Looks good to me!

**Pull request merged**